### ISSUE TITLE
Can fall into cave after spawning

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#12410]
##### Minetest version
```
5.6.0-dev-381f84ee2
```
##### Summary
Sometimes, the player spawn can be pretty unlucky, causing you to fall into a cave after spawning.

##### Steps to reproduce
This occured in a fresh, unmodified map and happens on the first spawn but also happens on subsequent spawns with high probability (not 100% probability tho).

To reproduce this, use:

* Mapgen: v6
* Seed: 15739225189199368382

If this does not happen in the first try, just do `/kill` and respawn.

I noticed I reliably spawn within the bounds of XY (-1,-1) to (1,1) on every spawn. The cave happens to be right below that.

**Comment by Zughy** [comment]
Does it happen only with V6? Also, what's MT policy about older mapgens? Do we still want to support them? (As I guess it's just the inferior version of V7, and the core dev focusing on them is gone anyway)

**Comment by rollerozxa** [comment]
> Does it happen only with V6? Also, what's MT policy about older mapgens? Do we still want to support them? (As I guess it's just the inferior version of V7, and the core dev focusing on them is gone anyway)

From the Minetest wiki about [map generators](https://wiki.minetest.net/Map_generator):
> **Important: v5 and v6 are not legacy map generators.** “v5”, “v6” and “v7” are names, not version numbers. All mapgens included in Minetest are officially supported and updated.

**Comment by Zughy** [comment]
> “v5”, “v6” and “v7” are names, not version numbers

yeah, that's so counterintuitive. Anyway, Paramat is gone, so core devs might want to reconsider that statement, as I don't think there is anyone specialised in mapgen (and as far as I know there are members of the community working on a better mapgen)

**Comment by Zughy** [comment]
And yep, that's not great. It's not like I'm underground but the only way to escape is to dig enough dirt to build a column

![imagen](https://user-images.githubusercontent.com/63455151/174178537-40249b89-caab-4857-a236-1c77514d2f4c.png)

**Comment by Vort** [comment]
Not sure if I encountered the same problem or not.
I was spawned midair when joined oysterity Mineclonia server.
Spawn locations on multiplayer servers can be very different from "unmodified terrain" version.
Such level can only be starting point of location search, not the final one.
